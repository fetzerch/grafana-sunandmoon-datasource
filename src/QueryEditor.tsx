import defaults from 'lodash/defaults';
import map from 'lodash/map';

import React, { PureComponent } from 'react';
import { InlineFormLabel, Select } from '@grafana/ui';
import { QueryEditorProps, SelectableValue } from '@grafana/data';
import { SunAndMoonDataSource } from './DataSource';
import { SunAndMoonDataSourceOptions, SunAndMoonQuery, sunAndMoonDefaultQuery, sunAndMoonMetrics } from './types';

export type Props = QueryEditorProps<SunAndMoonDataSource, SunAndMoonQuery, SunAndMoonDataSourceOptions>;

const metrics = map(sunAndMoonMetrics, (label, metric) => {
  return { label: label, value: metric };
});

export class QueryEditor extends PureComponent<Props> {
  onMetricChanged = (selected: SelectableValue<string>) => {
    const { onChange, query, onRunQuery } = this.props;
    onChange({ ...query, target: selected.value });
    onRunQuery();
  };

  render() {
    const query = defaults(this.props.query, sunAndMoonDefaultQuery);
    return (
      <div className="gf-form">
        <InlineFormLabel>Metric</InlineFormLabel>
        <Select
          className="metric"
          options={metrics}
          value={query.target}
          onChange={this.onMetricChanged}
          placeholder="Metric"
        />
      </div>
    );
  }
}
