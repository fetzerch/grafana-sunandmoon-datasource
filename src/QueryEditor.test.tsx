import React from 'react';
import { shallow } from 'enzyme';
import { QueryEditor } from './QueryEditor';

const setup = (propsOverrides?: object) => {
  const props: any = {
    query: {
      target: 'sun_altitude',
    },
  };

  Object.assign(props, propsOverrides);
  return shallow(<QueryEditor {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should update target on change', () => {
    const onChange = jest.fn();
    const onRunQuery = jest.fn();
    const wrapper = setup({ onChange: onChange, onRunQuery: onRunQuery });
    const metricSelect = wrapper.find('.metric');

    metricSelect.simulate('change', { value: 'moon_altitude' });
    expect(onChange).toHaveBeenCalledWith({ target: 'moon_altitude' });
    expect(onRunQuery).toHaveBeenCalled();
  });
});
