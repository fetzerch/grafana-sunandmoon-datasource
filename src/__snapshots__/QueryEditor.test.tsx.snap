// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="gf-form"
>
  <FormLabel>
    Metric
  </FormLabel>
  <Select$1
    className="metric"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Moon illumination",
          "value": "moon_illumination",
        },
        Object {
          "label": "Moon altitude",
          "value": "moon_altitude",
        },
        Object {
          "label": "Moon azimuth",
          "value": "moon_azimuth",
        },
        Object {
          "label": "Moon distance",
          "value": "moon_distance",
        },
        Object {
          "label": "Sun altitude",
          "value": "sun_altitude",
        },
        Object {
          "label": "Sun azimuth",
          "value": "sun_azimuth",
        },
      ]
    }
    placeholder="Metric"
    value="sun_altitude"
  />
</div>
`;
